packages/core/src/fhir-deduplication/deduplicate-fhir.ts

```diff
@@ -29,350 +30,285 @@ import { createRef } from "./shared";
-      // Extract all other resources
-      const entriesArray = resources && Array.isArray(resources) ? resources : [resources];
-      const entriesFlat = entriesArray.flatMap(v => v || []);
-      const entriesWithDeduplicatedReferences = entriesFlat.map(removeDuplicateReferences);
-      deduplicatedEntries.push(...entriesWithDeduplicatedReferences);
-    }
-  }
 
-  const deduplicatedNoDangling = removeResourcesWithDanglingLinks(
-    deduplicatedEntries,
+  // Combine all dangling references
+  const danglingLinks = new Set([
+    ...medicationsResult.danglingReferences,
+    ...medAdminsResult.danglingReferences,
+    ...medRequestResult.danglingReferences,
+    ...medStatementResult.danglingReferences,
+    ...practitionersResult.danglingReferences,
+    ...conditionsResult.danglingReferences,
+    ...allergiesResult.danglingReferences,
+    ...encountersResult.danglingReferences,
+    ...diagReportsResult.danglingReferences,
+    ...immunizationsResult.danglingReferences,
+    ...proceduresResult.danglingReferences,
+    ...obsSocialResult.danglingReferences,
+    ...obsLabsResult.danglingReferences,
+    ...obsVitalsResult.danglingReferences,
+    ...obsOthersResult.danglingReferences,
+    ...locationsResult.danglingReferences,
+    ...organizationsResult.danglingReferences,
+    ...relatedPersonsResult.danglingReferences,
+    ...famMemHistoriesResult.danglingReferences,
+    ...coveragesResult.danglingReferences,
+  ]);
+
+  // Combine all the remaining replacementMaps into one map
+  const combinedReplacementMap = new Map<string, string>([
+    ...medAdminsResult.refReplacementMap,
+    ...medRequestResult.refReplacementMap,
+    ...medStatementResult.refReplacementMap,
+    ...practitionersResult.refReplacementMap,
+    ...conditionsResult.refReplacementMap,
+    ...allergiesResult.refReplacementMap,
+    ...encountersResult.refReplacementMap,
+    ...diagReportsResult.refReplacementMap,
+    ...immunizationsResult.refReplacementMap,
+    ...proceduresResult.refReplacementMap,
+    ...obsSocialResult.refReplacementMap,
+    ...obsLabsResult.refReplacementMap,
+    ...obsVitalsResult.refReplacementMap,
+    ...obsOthersResult.refReplacementMap,
+    ...locationsResult.refReplacementMap,
+    ...organizationsResult.refReplacementMap,
+    ...relatedPersonsResult.refReplacementMap,
+    ...famMemHistoriesResult.refReplacementMap,
+    ...coveragesResult.refReplacementMap,
+  ]);
+
+  resourceArrays = replaceResourceReferences(resourceArrays, combinedReplacementMap);
+
+  // Remove resources with dangling links
+  const { updatedResourceArrays, deletedRefs } = removeResourcesWithDanglingLinks(
+    resourceArrays,
     danglingLinks
   );
+  resourceArrays = updatedResourceArrays;
 
-  const compositionsNoDangling = removeResourcesWithDanglingLinks(
-    resourceArrays.compositions,
-    danglingLinks
+  const { updatedResourceArrays: updatedResourceArrays2 } = removeResourcesWithDanglingLinks(
+    resourceArrays,
+    deletedRefs
```

Comment:
get dangling links inside compositions from medication related resources we deleted in the previous call to `removeResourcesWithDanglingLinks`

Reply:
put this in the code comment please